common/src/test/java/org/opensearch/ml/common/agent/MLAgentTest.java

```diff
@@ -46,7 +46,7 @@ public void constructor_NullName() {
             MLAgentType.CONVERSATIONAL.name(),
             "test",
             new LLMSpec("test_model", Map.of("test_key", "test_value")),
-            List.of(new MLToolSpec("test", "test", "test", Collections.EMPTY_MAP, false)),
+            List.of(new MLToolSpec("test", "test", "test", Collections.EMPTY_MAP, false, Collections.EMPTY_MAP)),
```

Comment:
```suggestion
            List.of(new MLToolSpec("test", "test", "test", Collections.EMPTY_MAP, false, Collections.emptyMap())),
```
Use `emptyMap()` rather than `EMPTY_MAP` to get built in type validation.  This will probably create unchecked cast warnings.  (Same comment multiple times below.)

Reply:
Makes sense, changed.